lib/avrora/storage/registry.ex

```diff
@@ -93,6 +94,35 @@ defmodule Avrora.Storage.Registry do
   @spec configured?() :: true | false
   def configured?, do: !is_nil(registry_url())
 
+  defp extract_references(response) do
+    case Map.fetch(response, "references") do
+      {:ok, references} ->
+        references_map =
+          Enum.reduce(references, %{}, fn cr, map ->
+            get_reference_schema(map, cr["subject"])
+          end)
+
+        {:ok, references_map}
+
+      :error ->
+        {:ok, :schema_without_references}
+    end
+  end
```

Comment:
By using throw I mean that we need to exit from the inner loop with error, but not throw it to outside, `throw` it's just a control flow in this case.

I would like to suggest something like this

```suggestion
  defp extract_references(response) do
    references =
      response
      |> Map.get("references", [])
      |> Enum.reduce(%{}, fn reference, memo ->
        case get(reference["subject"]) do
          {:ok, schema} -> Map.put(memo, schema.full_name, schema.json)
          {:error, error} -> throw(error)
        end
      end)
  
    {:ok, references}
  catch
    {:error, :unknown_subject} -> {:error, :unknown_reference_subject}
    error -> {:error, error}
  end
```